text/0000-eliminate-finalize.md

```diff
@@ -0,0 +1,142 @@
+
+## Slate Format
+
+A slate is now only needed for passing information from the sender to receiver. It must include the following data:
+
+* `tx.amount`
+* `tx.offset`
+* `tx.kernel_features`
+* `tx.fee`
+* `tx.inputs`
+* `receiver.change_output`
+* `receiver.public_excess` & `receiver.public_nonce` (to prevent the need for grinding)
+* `sender.public_excess`, `sender.public_nonce`, & `sender.partial_sig`
+* `time_to_live` (optional)
+
+TODO: Define JSON & binary formats
+TODO: Mention QR codes
+
+## Backward Compatibility
+
+TODO: Decide how to handle versioning. In theory, this should be simple. If given an old slatepack address, use the old slate format. If given a new, one-time address, then the new slate format is supported.
+
+## Invoices
+
+To send an invoice to a payer, the recipient provides the payer with a newly generated one-time address, an amount, and any additional fields we decide to support (memo, refund address, expiration, etc). The payer, upon approving payment of the invoice, simply follows the standard 2-step send process described in this RFC. There's no need for a special workflow.
+
+## Security
+
+### Nonce Reuse
+
+Nonces should be randomly generated by the receiving wallet, rather than derived from seed, since reuse of a nonce could lead to loss of funds. Receiving wallets should keep track of all one-time addresses (excesses & nonces) they generate in a list (`address_list`), along with an indicator for whether they have been used yet.
+
+Upon receipt of a partial transaction (slate), the receiver should obtain a lock on `address_list` and then check whether the address has been used before. If it has not, it's safe to finalize the transaction, but before broadcasting, the address should be marked as used in the `address_list`, and then the lock can be safely released.
+
+### Play Attacks
+
+TODO: Describe play attacks, and finish this section.
+
+Any inputs sent from a wallet to a one-time address should be considered as sent. They should not be cancelable, however, the sender should have the option to add additional inputs and/or increase the fee (equivalent of bitcoin's RBF).
+
+Receiver should be unable to claim a problem with the transaction. If they do, it's trivial to show them the slate that was sent, and prove that it's a valid partial transaction containing everything necessary for the receiver to finalize.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Addresses are significantly longer (170 characters).
+
+Using one-time addresses increases the risk of "Play" attacks [2]. Recommendations for dealing with these have been discussed in the above "Play attacks" section.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+An alternative approach to simplifying transaction building is to support non-interactive transactions [3].
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+* Can this be used to simplify sending to a multisig address?
```

Comment:
Interesting.... We should explore this, it's one of the open things with Slatepacks as well. If we could ascertain that this flow would make multisigs (and maybe also atomic swaps?) easier, it could be a powerful statement.